### PR TITLE
Reimplement the :firstn strategy

### LINKED ISSUE #30: Reintroduce `pinthreads(:firstn)`
Doing essentially `pinthreads(cpuids_all()[1:nthreads()])`.

### PR DESCRIPTION
**Pull request by JamesWrigley**
Fixes #30.

### CODE REVIEW

File: src/pinning.jl
Comment (carstenbauer) [inline src/pinning.jl:143]: This is only correct if the first n cpu ids are indeed 0,1,2,3,... etc. which isn't the case on all systems. Instead, `:firstn` pins to the first n cpuids (as reported by lscpu, which is the basis for everything in ThreadPinning.jl).

File: src/pinning.jl
Comment (carstenbauer) [inline src/pinning.jl:130]: Might need a bounds check. For example, `nthreads()` could be larger than the number of available cpu ids.

**Review by carstenbauer — COMMENTED** [review]
Thanks for the PR! See my comments :)

File: src/pinning.jl
Comment (JamesWrigley) [reply src/pinning.jl]: Right, I keep forgetting that :facepalm: Fixed in e9b3810.

**Comment by JamesWrigley** [comment]
BTW Github let's you explicitly approve a PR, if you do that or comment your approval, I'll squash the fixup comments before merging :) I meant to do that for #17.

**Review by carstenbauer — APPROVED** [review]

**Pull request merged**
